mautic/mautic
# Allow contact to set DNC flag in preferences center

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | N
| New feature? | N
| Automated tests included? | N
| Related user documentation PR URL | https://github.com/mautic/mautic-documentation/pull/88
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | N
| Deprecations? | N

[//]: # ( Required: )
#### Description:
When preference center is activated there is no option for user to unsubscribe from all marketing messages. This PR allows user to set Do Not Contact flag by clicking link in subscription center. This is optional and can be turned off in Mautic configuration.

![image](https://user-images.githubusercontent.com/462477/140615425-249d217d-1b49-45f3-a6f6-b74e54cfdd5b.png)

#### Steps to test this PR:
1. Load up [this PR](https://mautibox.com)
2. Activate Preferences center and Do Not Contact option in Mautic settings
3. Send email to user with unsubscribe link
4. Open preferences center (click unsubscribe link)
5. Click Unsubscribe from all marketing messages link
6. Contact should have DNC flag on
